modules/geo-layers/src/quadkey-layer/quadkey-layer.ts

```diff
@@ -28,14 +28,16 @@ export default class QuadkeyLayer<DataT = any, ExtraProps extends {} = {}> exten
   static defaultProps = defaultProps;
 
   indexToBounds(): Partial<GeoCellLayer['props']> | null {
-    const {data, getQuadkey} = this.props;
+    const {data, extruded, getQuadkey} = this.props;
+    // To avoid z-fighting reduce polygon footprint when extruding
+    const coverage = extruded ? 0.99 : 1;
 
     return {
       data,
       _normalize: false,
       positionFormat: 'XY',
 
-      getPolygon: (x: DataT, objectInfo) => getQuadkeyPolygon(getQuadkey(x, objectInfo))
```

Comment:
Does this work when you switch between extruded and non-extruded? Needs an update trigger?